### DISCUSSION TITLE
Wildcard SMARTS [*] atom issue, are hydrogen atoms not considered wildcard atoms?

### DISCUSSION THREAD

**Discussion by DoWeHaveChemistry** [rdkit/rdkit#6629] [Q&A]
hi, trying to get my code to recognise a pattern where a carbon is attached to a chlorine atom and potentially any other three atoms during a substructure search.

This code:
![image](https://github.com/rdkit/rdkit/assets/142227797/b88f2008-e2b5-4ffb-b0e5-ce647667a592)

with the output: 
CC(Cl)(C)C
<rdkit.Chem.rdchem.Mol object at 0x000001E9091A1C40>
[*]C([*])([*])Cl
<rdkit.Chem.rdchem.Mol object at 0x000001E9091A1CB0>
[]

whearas the above code changing the smi variable to smi = 'CC(Cl)(C)C'  :

gives the output: 
CC(Cl)(C)C
<rdkit.Chem.rdchem.Mol object at 0x000001E9091A1C40>
[*]C([*])([*])Cl
<rdkit.Chem.rdchem.Mol object at 0x000001E9091A1CB0>
[([0, 1, 3, 4, 2], [0, 2, 3, 1])]

It seems perhaps the hydrogen atoms are not considered wildcard atoms in the first example?

**Comment by ptosco** [comment]
Please paste your code as text, not as image, or people willing to answer your question will have to retype your code to reproduce your case.

What you are observing is due to the fact that hydrogen atoms are implicit in a molecule constructed from SMILES, i.e. they are not real atoms that you can match in a substructure search. You either need to use SMARTS queries on the parent heavy atom to specify the number of implicit Hs you want to match, or add explicit Hs to your molecule with `Chem.AddHs()` before doing the substructure match.

**Reply by DoWeHaveChemistry** [reply]
Thank you kindly. I tried to paste my code raw from jupyter in but the formatting was unusable it made some text and some was a recognised copyable format that retains the code like formatting. Ideally i wanted it to be all in the latter so someone could just click it and copy it but I had no luck (I'm new to the community).

**Reply by ptosco** [reply]
To preserve code-like formatting you need to paste your code between three-backtick guards:
```
three backticks before, three backticks after
```

**Reply by DoWeHaveChemistry** [reply]
Many thanks Paolo!
